binary_transparency/firmware/internal/verify/bundle.go

```diff
@@ -44,42 +45,57 @@ func BundleForUpdate(bundleRaw, fwHash []byte, dc api.LogCheckpoint, cpFunc Cons
 		return proofBundle, fwMeta, fmt.Errorf("firmware update image hash does not match metadata (0x%x != 0x%x)", got, want)
 	}
 
-	cProof, err := cpFunc(dc.TreeSize, proofBundle.Checkpoint.TreeSize)
+	pcNote, err := note.Open(proofBundle.Checkpoint, note.VerifierList(cpSigVerifier))
+	if err != nil {
+		return proofBundle, fwMeta, fmt.Errorf("failed to open the device checkpoint: %w", err)
+	}
+	pc := &api.LogCheckpoint{Envelope: proofBundle.Checkpoint}
+	if err := pc.Unmarshal([]byte(pcNote.Text)); err != nil {
+		return proofBundle, fwMeta, fmt.Errorf("failed to unmarshal the device checkpoint: %w", err)
+	}
+
+	cProof, err := cpFunc(dc.Size, pc.Size)
 	if err != nil {
 		return proofBundle, fwMeta, fmt.Errorf("cpFunc failed: %q", err)
 	}
 
 	// Verify the consistency proof between device and bundle checkpoint
-	if dc.TreeSize > 0 {
+	if dc.Size > 0 {
 		lv := NewLogVerifier()
-		if err := lv.VerifyConsistencyProof(int64(dc.TreeSize), int64(proofBundle.Checkpoint.TreeSize), dc.RootHash, proofBundle.Checkpoint.RootHash, cProof); err != nil {
+		if err := lv.VerifyConsistencyProof(int64(dc.Size), int64(pc.Size), dc.Hash, pc.Hash, cProof); err != nil {
 			return proofBundle, fwMeta, fmt.Errorf("failed verification of consistency proof %w", err)
 		}
 	}
 	return proofBundle, fwMeta, nil
 }
 
 // BundleConsistency verifies the log checkpoint in the bundle is consistent against a given checkpoint (e.g. one fetched from a witness).
-func BundleConsistency(pb api.ProofBundle, rc api.LogCheckpoint, cpFunc ConsistencyProofFunc) error {
+func BundleConsistency(pb api.ProofBundle, rc api.LogCheckpoint, cpFunc ConsistencyProofFunc, cpSigVerifier note.Verifier) error {
```

Comment:
cpSigVerifier in this specific context easily leads to expanding cp to "consistency proof". I wonder if `c` or `checkpoint` or `log` would be better prefixes?

Reply:
Yeah, I think I'll move to `logSigVerifier` everywhere